### PR TITLE
docs: Updated sync and dedupe command docs #4429

### PR DESCRIPTION
**Pull request by eharris**
Update the docs regarding sync and dedupe commands with info regarding handling duplicate objects.  See ticket #4429.

### CODE REVIEW

File: docs/content/commands/rclone_dedupe.md [hunk 1]
```diff
@@ -13,8 +13,8 @@ Interactively find duplicate files and delete/rename them.
 
 
 By default `dedupe` interactively finds duplicate files and offers to
-delete all but one or rename them to be different. Only useful with
-Google Drive which can have duplicate file names.
+delete all but one or rename them to be different.  Only some backends
```
Comment (ncw) [inline docs/content/commands/rclone_dedupe.md:16]: This is an autogenerated file - the file which needs editing is in `cmd/dedupe`

File: docs/content/commands/rclone_sync.md [hunk 2]
```diff
@@ -15,13 +15,15 @@ Make source and dest identical, modifying destination only.
 Sync the source to the destination, changing the destination
 only.  Doesn't transfer unchanged files, testing by size and
 modification time or MD5SUM.  Destination is updated to match
-source, including deleting files if necessary.
+source, including deleting files if necessary (except duplicate
```
Comment (ncw) [inline docs/content/commands/rclone_sync.md:18]: This is an autogenerated file - the file which needs editing is in `cmd/sync`

File: docs/content/commands/rclone_dedupe.md [hunk 3]
```diff
@@ -13,8 +13,8 @@ Interactively find duplicate files and delete/rename them.
...
+(e.g. Google Drive, Opendrive) have the ability to store duplicate file names.
```
Comment (ncw) [inline docs/content/commands/rclone_dedupe.md:17]: I'm not sure the Opendrive isn't a bug?

Google Drive and Mega definitely can and have rclone dedupe support. Opendrive doesn't have dedupe support yet so can't actually run dedupe.

File: docs/content/commands/rclone_sync.md [hunk 4]
```diff
@@ -32,6 +34,8 @@ If dest:path doesn't exist, it is created and the source:path contents
 go there.
 
 **Note**: Use the `-P`/`--progress` flag to view real-time transfer statistics
+**Note**: Use the `rclone dedupe` command to deal with "Duplicate object found in destination" errors.
```
Comment (ncw) [inline docs/content/commands/rclone_sync.md:37]: I think this needs a newline before it otherwise it will be rolled into a single paragraph

Comment (ncw) [inline docs/content/commands/rclone_sync.md:37]: This should probably say ... to deal with "Duplicate object/directory found in source/destination - ignoring" errors as the error message can be any of those.

File: docs/content/commands/rclone_dedupe.md (hunk 3 above)
Comment (eharris) [reply docs/content/commands/rclone_dedupe.md]: I don't know for sure if Opendrive is a bug or unsupported feature, but dupes and `rclone dedupe` (of files at least) seems to work.

File: cmd/sync/sync.go [hunk 5]
```diff
@@ -46,6 +48,9 @@ If dest:path doesn't exist, it is created and the source:path contents
 go there.
 
 **Note**: Use the ` + "`-P`" + `/` + "`--progress`" + ` flag to view real-time transfer statistics
+
+**Note**: Use the `rclone dedupe` command to deal with "Duplicate object/directory found in source/destination - ignoring" errors.
```
Comment (ncw) [inline cmd/sync/sync.go:52]: This needs the ` dealing with like the ones in the line above

File: docs/content/commands/rclone_dedupe.md (hunk 3 above)
Comment (ivandeex) [reply docs/content/commands/rclone_dedupe.md]: I removed "Opendrive" from this line

**Comment by ivandeex** [comment]
@eharris If you set the check "Allow maintainers to edit this PR" on the PR page, please let me know. I will be able to rebase for you then.

File: docs/content/commands/rclone_dedupe.md (hunk 3 above)
Comment (ivandeex) [reply docs/content/commands/rclone_dedupe.md]: @eharris I propose to keep Google Drive and keep questionable Opendrive for future. It really isn't importand at all.

Comment (ivandeex) [reply docs/content/commands/rclone_dedupe.md]: This small change hangs for too long. Let's have it in.

**Comment by ivandeex** [comment]
The last @eharris reply is from July 2020. Marking as draft until this PR shows signs of life.

**Comment by eharris** [comment]
I pushed an update to fix the merge conflict over a week ago. What else is needed?

**Comment by ivandeex** [comment]
See inline comment thread

**Comment by ivandeex** [comment]
@ncw 

We have a severe communication problem here.
Here is the "discussion" that started in June 2020 in the inline thread :-/

@ncw: `Opendrive doesn't have dedupe support yet so can't actually run dedupe.`
@eharris: `I don't know for sure if Opendrive is a bug or unsupported feature, but dupes and rclone dedupe (of files at least) seems to work.`

7 months later!!!!

@ivandeex: `I propose to keep Google Drive and keep questionable Opendrive for future. It really isn't importand at all.`
@eharris (out of thread): `I pushed an update to fix the merge conflict over a week ago. What else is needed?`

Obviously we have a communication problem.
I (@ivandeex) tried to push a patch that removes questionable word from docs but got permission denied.
Obviously we have a permission problem on top of communication problem.

I'm marking this thread with `regional` label which I normally use for threads with broken communication (usually due to bad English but probably for other reasons, e.g. see many threads with zhucаn where they continuously copy-paste the same sentence slightly edited instead of answering questions)

P.S. The whole patch here is less than 10 lines of code............

File: docs/content/commands/rclone_dedupe.md (hunk 3 above)
Comment (eharris) [reply docs/content/commands/rclone_dedupe.md]: Dupes are still a problem on Opendrive, and the `rclone dedupe` command does currently work (as I made clear in the inline thread) so I'm unclear as to why you want to remove Opendrive from being mentioned as a backend that has this issue.  What exactly are the changes you want me to make?

**Comment by ivandeex** [comment]
@ncw
Please review this docs-only change.
(The author insists on having Opendrive)

**Comment by eharris** [comment]
To be clear, I just think it's important to try to dispel the myth that dupes is only a problem for Google Drive, which is not the case.

**Comment by ivandeex** [comment]
@animosity22 

In https://github.com/rclone/rclone/issues/4429#issuecomment-660263357 you wrote:

> > Whether the endpoints supports duplicates may not be known, but I would also argue that it doesn't matter since the behavior is the same. That is, even if the source and/or destination supports dupes, if the source doesn't contain any dupes for a particular file, then it should still use the methodology for case 2.
> >
> > [snip]
>
> That makes sense to me as well and a very good descriptions.
> This is only for OpenDrive and Google Drive I think as well or is there another one that would be impacted?

Now the author proposes this help string as part his patch (it's small):
```
Only some backends (e.g. Google Drive, Opendrive) have the ability to store duplicate file names.
```

Idea: review the patch and make a verdict

**Comment by animosity22** [comment]
That looks good to me. I primarily use Google so not quite as familiar with the others, but adding more backends and listing them out seems proper to me.

**Comment by ncw** [comment]
OK I will merge this now. Thanks for your patience @eharris :-)

**Pull request merged**